### PR TITLE
FIX update API changes get_result and evaluate_result

### PR DESCRIPTION
**Pull request by mathurinm**

### CODE REVIEW

**Comment by mathurinm** [comment]
Also updated (simplified) the scipy solver because the fortran code was deleted in favor of a pure python implem in version 1.12: https://github.com/scipy/scipy/pull/18570

File: solvers/nnls_scipy.py [hunk 1]
```diff
@@ -1,37 +1,22 @@
...
     install_cmd = 'conda'
-    requirements = ['scipy']
+    requirements = ['scipy>=1.12']
 
     def set_objective(self, X, y, fit_intercept=False):
         self.X, self.y = X, y
         self.fit_intercept = fit_intercept
 
     def run(self, n_iter):
-        m, n = self.X.shape
```
Comment (Badr-MOUFAD) [inline solvers/nnls_scipy.py:21]: Any idea why we wrapped `nnls` in try expect block?

Comment (mathurinm) [reply solvers/nnls_scipy.py:21]: The public `nnls` solver raises an issue it it does not converge: https://github.com/scipy/scipy/blob/main/scipy/optimize/_nnls.py#L92

this prevents us from sampling the curve. hence we have to use private methods

**Review by Badr-MOUFAD — APPROVED** [review]
LGTM!

though pining scipy `v1.12` might break others' code as it is the most recent version

**Pull request merged**